.github/workflows/ci.yml

```diff
@@ -12,7 +12,98 @@ env:
   GOTESTSUM_CMD: "gotestsum --format standard-verbose --debug --"
   GOTESTSUM_CMD_RAW: "gotestsum --format standard-verbose --debug --raw-command -- go tool test2json -t"
 
+  LCOW_ARTIFACT_PROJECT: "ContainerPlatform"
+  LCOW_ARTIFACT_FEED: "ContainerPlat-Dev"
+  LCOW_ARTIFACT_NAME: "azurelinux-uvm"
+  LCOW_ARTIFACT_VERSION: "0.0.13"
+  LINUX_BOOT_FILES_PATH: ${{ github.workspace }}/LinuxBootFiles
+
 jobs:
+  create-linux-boot-files:
+    runs-on: [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ubuntu-22]
+    permissions:
+      id-token: write   # This is required for OIDC login (azure/login) to succeed
+      contents: read    # This is required for actions/checkout to succeed
+    steps:
+    - name: Checkout hcsshim
+      uses: actions/checkout@v4
+      with:
+        show-progress: false
+
+    - name: Azure OIDC Login
+      uses: azure/login@v2
+      with:
+        client-id: "930a0428-2b45-4cf9-9afe-b81bde516504"
+        tenant-id: "72f988bf-86f1-41af-91ab-2d7cd011db47"
+        allow-no-subscriptions: true
+
+    - name: Download artifact from feed
+      uses: azure/cli@v2
+      with:
+        azcliversion: latest
+        inlineScript: |
+          az extension add --name azure-devops
+          export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
+
+          az artifacts universal download \
+          --organization "https://msazure.visualstudio.com/" \
+          --project ${{ env.LCOW_ARTIFACT_PROJECT }} \
+          --scope project \
+          --feed  ${{ env.LCOW_ARTIFACT_FEED }} \
+          --name  ${{ env.LCOW_ARTIFACT_NAME }} \
+          --version  ${{ env.LCOW_ARTIFACT_VERSION }} \
+          --path ./downloaded_artifacts
+
+    - name: Show downloaded lcow artifacts
+      run: find ./downloaded_artifacts -maxdepth 3 -ls
+
+    - name: Create directory for storing linux boot files
+      run: mkdir -p ${{ env.LINUX_BOOT_FILES_PATH }}/
+
+    - name: Copy Linux kernel and rootfs tar files
+      run: |
+        mv ./downloaded_artifacts/LinuxBootFiles/kernel ${{ env.LINUX_BOOT_FILES_PATH }}/
+        mv ./downloaded_artifacts/LinuxBootFiles/vmlinux ${{ env.LINUX_BOOT_FILES_PATH }}/
+        mv ./downloaded_artifacts/rootfs-*.tar.gz .
+
+    - name: Create a new rootfs.vhd and initrd.img containing the local Linux-GCS
+      run: |
+        # Update package on Ubuntu
+        sudo apt-get update
+        sudo apt-get install -y make gcc binutils linux-headers-generic \
+          libarchive-tools btrfs-progs libseccomp-dev pkg-config cpio
+
+        chmod a+x ${{ github.workspace }}/hack/catcpio.sh
+
+        # Find the full file name for rootfs tar
+        ROOTFS_TAR=$(ls rootfs-*.tar.gz | head -n 1)
+        echo "The full file name is $ROOTFS_TAR"
+
+        make clean
+        sudo make BASE=${{ github.workspace }}/$ROOTFS_TAR rootfs
```

Comment:
Can we also set "KMOD=1" as an arg to `make`?

Reply:
Updated!